packages/webrtc-core/src/media/track.ts

```diff
@@ -0,0 +1,198 @@
+/* eslint-disable no-underscore-dangle */
+import { EventMap } from 'typed-emitter';
+import { EventEmitter } from '../event-emitter';
+import { logger } from '../util/logger';
+import { MediaStreamTrackKind } from '../peer-connection';
+
+export enum Events {
+  Ended = 'ended',
+  Muted = 'muted',
+  /**
+   * Fires when the published state of a LocalTrack changes.  A value of 'true' indicates the
+   * track has been published, and 'false' indicates that it is no longer published.  Tracks are
+   * unpublished by default, and must be published explicitly (no event will be fired to indicate
+   * the initial state of 'unpublished').
+   */
+  PublishedStateUpdate = 'published-state-update',
```

Comment:
does this belong here? only LocalTrack has setPublished() method, not the Track

Reply:
if we end up having remote track extending track  that should not have publish function so kept it on local track

Reply:
publish state doesn't make any sense for a remote track?